src/System.Device.Gpio/System/Device/Gpio/IGpioController.cs

```diff
@@ -2,19 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Device.Gpio;
-
-namespace Iot.Device.CharacterLcd
+namespace System.Device.Gpio
 {
     public interface IGpioController : IDisposable
     {
         void OpenPin(int pinNumber, PinMode mode);
         void ClosePin(int pinNumber);
         void SetPinMode(int pinNumber, PinMode mode);
         void Write(int pinNumber, PinValue value);
-        void Write(ReadOnlySpan<(int pin, PinValue value)> pinValues);
+        void Write(ReadOnlySpan<PinValuePair> pinValues);
         PinValue Read(int pinNumber);
-        void Read(Span<(int pin, PinValue value)> pinValues);
+        void Read(Span<PinValuePair> pinValues);
```

Comment:
Per my other comment, I would keep this as ValueTuple for now.

Reply:
The problem with that is that you cannot stack allocate. ValueTuple is generic.

Reply:
The goal here is to turn this into `Read(params Span<PinValuePair> pinValues)`.

Reply:
I'm wondering if we should come up with something more generic since if I want to control say DC motor with 1 pin on the expander and 1 on the gpio now I have to have 2 GpioControllers. Perhaps we should have GpioController have some option of adding virtual pins and be able to get the virtual pin number from the expander.

Reply:
That would require a whole different set of changes, since it would require a GpioPin object instead. If you had that, then you would be able to just pass those in without the device needing to know who is providing them. I think that if we wanted to discuss about that, it should be logged into a new issue.

Reply:
It's not a separate issue, but related was somewhat mentioned here:
https://github.com/dotnet/iot/issues/125#issuecomment-460009493